File: backend/src/main/java/com/showmantra/services/MovieService.java
package com.showmantra.services;

import com.showmantra.dtos.MovieRequest;
import com.showmantra.entities.Movie;
import com.showmantra.repositories.MovieRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class MovieService {

    private final MovieRepository movieRepository;

    public Movie createMovie(MovieRequest request) {
        Movie movie = Movie.builder()
                .title(request.title())
                .durationMinutes(request.durationMinutes())
                .language(request.language())
                .genre(request.genre())
                .releaseDate(request.releaseDate())
                .build();
        return movieRepository.save(movie);
    }

    public List<Movie> getAllMovies() {
        return movieRepository.findAll();
    }

    public List<Movie> searchMovies(String title) {
        return movieRepository.findByTitleContainingIgnoreCase(title);
    }
}

File: frontend/src/services/movieService.ts
import { api } from './api';

export interface Movie {
  id: number;
  title: string;
  genre: string;
  language: string;
  image?: string;
  description?: string;
  durationMinutes?: number;
  releaseDate?: string;
}

export const movieService = {
  getAllMovies: async (cityId?: number): Promise<Movie[]> => {
    const url = cityId ? `/movies?cityId=${cityId}` : '/movies';
    const response = await api.get(url);
    // The backend returns an array of Movie entities
    return response.data;
  },
  searchMovies: async (query: string): Promise<Movie[]> => {
    const response = await api.get(`/movies/search?q=${encodeURIComponent(query)}`);
    return response.data;
  }
};
